### PR TITLE
[UR][L0] Check Global Mem Size as Limit for Free Memory

### PR DESCRIPTION
**Pull request by nrspruit**

### CODE REVIEW

**Comment by nrspruit** [comment]
pre-commit checks in the draft PR https://github.com/intel/llvm/pull/11379

File: source/adapters/level_zero/device.cpp
Comment (jandres742) [inline source/adapters/level_zero/device.cpp:672]: why is that we want to return the min?

Comment (nrspruit) [reply source/adapters/level_zero/device.cpp:672]: We want to return the size up to global memsize, not larger than.

The Free memory could exceed the Global Memory size since Global Memory Size is a calculation, but to avoid OOM condition, we want to have the minimum of the two ie GlobalMemSize > Free Memory, then return Free Memory as the Free Memory Available, but if GlobalMemSize < Free Memory, then return the Global Memory Size as the max "free memory" available.

**Comment by jandres742** [comment]
Changes looking good. We will wait for pre-merge results from https://github.com/intel/llvm/pull/11379 before approving.

**Review by jandres742 — APPROVED** [review]

**Comment by jandres742** [comment]
waiting for llvm testing before moving to merging.

**Comment by fabiomestre** [comment]
I have updated the target branch of this PR from the `adapters` branch to the `main` branch.
Development in UR is moving back to `main`. The `adapters` branch will soon be deleted.

**Comment by kbenzie** [comment]
I'm going to create a combined intel/llvm PR which will include this and a few other PR's so we can get things merged quicker.

**Pull request merged**
